newsletter/models.py

```diff
@@ -1,18 +1,52 @@
 from django.db import models
 from django.utils import timezone
+import uuid
+
+# Custom Manager for active subscribers
+class ActiveSubscriberManager(models.Manager):
+    def get_queryset(self):
+        return super().get_queryset().filter(is_active=True)
 
 
 # Create your models here.
 class NewsletterSubscriber(models.Model):
     created_at = models.DateTimeField(default=timezone.now, editable=False)
     updated_at = models.DateTimeField(auto_now=True)
     email = models.EmailField(
-        max_length=300,
-        null=False,
-        blank=False,
-        unique=True,
-        primary_key=True,
+        max_length=300, 
+        null=False, 
+        blank=False, 
+        unique=True, 
+        primary_key=True
     )
+    
+    is_active = models.BooleanField(default=True)  # Track active/inactive status
+    
+    name = models.CharField(
+        max_length=200, 
+        null=True, 
+        blank=True
+    )  # Optional name field for personalization
+    
+    unsubscribe_token = models.CharField(
+        max_length=100, 
+        unique=True, 
+        default=uuid.uuid4
+    )  # Unique unsubscribe token
+    
+    subscription_date = models.DateTimeField(
+        default=timezone.now, 
+        editable=False
+    )  # Subscription timestamp
+    
+    last_sent = models.DateTimeField(
+        null=True, 
```

Comment:
Do we really need last_sent as a dedicated field ? @Sailesh-Singh 

Will we not be sending emails for everyone for email campaigns ?